### ISSUE TITLE
Wayland support

### ISSUE THREAD

**Issue by obv-mikhail** [obv-mikhail/InputBot#4]
Not sure what will be used for the back-end. Probably libinput.

**Comment by kitling** [comment]
Correct me if I'm wrong, but if you go through the kernel by using uinput then you wouldn't need to worry about X11/Wayland/etc...?

**Comment by obv-mikhail** [comment]
@kitling yes I think you are right. The best path might be to use uinput for generating events and using evdev for reading events to create hotkeys. Evdev has rust bindings as well https://github.com/myfreeweb/evdev .

**Comment by obv-mikhail** [comment]
Hotkeys should now work on Wayland since I made InputBot use Libinput for event listening. I wanted to use the uinput crate for simulating events but the crate fails to build.

**Comment by obv-mikhail** [comment]
Most functions of InputBot no longer rely on X11 APIs and should work on Wayland.

The parts left to port are the `is_pressed()` function for keys and the `move_abs()` function for the mouse cursor.

The `is_pressed` function can be ported by keeping track of all key states internally and simply returning their state instead of using `XQueryKeymap`.

**Comment by D01B0EB** [comment]
Not working at all.
Tested in Ubuntu 24.04 Wayland, Fedorа 43 Wayland.
Press F8 does nothing.

```rust
fn main() {
    F8Key.bind(move || {
        println!("{}{}", "Pressed F8Key!", F8Key.is_pressed().to_string());
    });
    println!("Started");
    inputbot::handle_input_events();
}
```

**UPDATE:**
Works when running as sudo.
User need permissions - group "input" |  udev rule | sudo.
